bulletroboy/exoforce/exoforce.py

```diff
@@ -466,3 +504,75 @@ def get_end_effectors_callback(self, request, response):
 			response.end_effectors.append(end_effector_msg)
 
 		return response
+
+	def decompose(self, link_id, collision_force, collision_direction):
+		"""Decomposes force applied to link in operator.
+		
+		Args:
+			link_id (int): Link id of the operator where the force is applied.
+			collision_force (float): Value of the applied force.
+			collision_direction array[3]: Force direction in world space coordinates.
+
+		Returns:
+		   	dict: Dictionary with the decomposed forces, the key is the tendon id.
+
+		"""
+		ef = self.decompose_force_link_to_ef(link_id)
+		self.get_logger().info("Force mapped to ef: " + ef)
+
+		forces, msg = decompose_force_ef_to_tendons(collision_force, collision_direction, self.get_ef_muscle_units(ef), self.force_decomp_params) if link_id is not None else {}
+		
+		if not forces:
+			self.get_logger().warn(f"Force was not decomposed: force[{collision_force}] ef[{ef}] [{msg}]")
+		
+		return forces
+		
+	def decompose_force_link_to_ef(self, link_id):
+		"""Decomposes a force applied to a link to an end effector.
+		
+		Args:
+			link_id (int): Index of the link where the force was applied.
+
+		Returns:
+			string: Name of the end effector.
+
+		"""
+		end_effector = None
+		for ef in self.end_effectors:
+			if link_id in self.end_effectors[ef]["decomp_to_link"]:
+				end_effector = ef
+				break
+		return end_effector
```

Comment:
why do you operate on names and ids at the same time? this is confusing
returning a name is not called decompose, the name of the function is misleading. all it does is finding a match to the endeffector

Reply:
The end effectors names are defined in the cage xml file:
https://github.com/Roboy/bulletroboy/blob/833c8775de086dab569622e48959147a60f4aa22/config/cageConfiguration.xml#L11
But we are receiving the link ids in the collision msg, so we are mapping this link ids to the end effector:
https://github.com/Roboy/bulletroboy/blob/833c8775de086dab569622e48959147a60f4aa22/config/conf.yaml#L29
Regarding the function name, that will be changed

Reply:
why not ids or string names only?